boards/arm/stm32h750b_dk/stm32h750b_dk.dts

```diff
@@ -81,3 +83,42 @@
 	current-speed = <115200>;
 	status = "okay";
 };
+
+&quadspi {
+	pinctrl-names = "default";
+	pinctrl-0 = <&quadspi_clk_pf10 &quadspi_bk1_ncs_pg6
+		     &quadspi_bk1_io0_pd11 &quadspi_bk1_io1_pf9
+		     &quadspi_bk1_io2_pf7 &quadspi_bk1_io3_pf6
+		     &quadspi_bk2_io0_ph2 &quadspi_bk2_io1_ph3
+		     &quadspi_bk2_io2_pg9 &quadspi_bk2_io3_pg14>;
+	flash-id = <1>;
```

Comment:
I'm not sure if this `flash-id` is needed

Reply:
It is used to select flash device in the Dual-flash mode, where 8 bits can be sent/received simultaneously by accessing two Flash memories in parallel.
However the   dev_data->hqspi.Init.DualFlash parameter is not set to QSPI_DUALFLASH_ENABLE. So `flash-id` property  is useless here (that configuration does not seem to be supported yet). Consequently you will not be able to access the two quad-Flash memories of the stm32h750 disco kit  in parallel without upgrading the flash_stm32_qspi driver.

Reply:
Ok, I removed it for now.

Reply:
Sorry, correcting with clarification :
The  dev_data->hqspi.Init.DualFlash parameter is not set by the driver meaning its value is 0  
 (ie QSPI_DUALFLASH_DISABLE) by default. So `flash-id` property is supposed to be used here to choose the flash memory to be accessed. The FlashID is initialized with 0 meaning the Flash 1 device (equivalent to  _not defined_ by the DTS or  `flash-id = <1>;` ).
If you want to access the second Flash 2 device (bank 2), then you should set `flash-id = <2>;`. I guess that has not been tested yet.

 The configuration with QSPI_DUALFLASH_ENABLE does neither seem to be supported/tested yet. That is for accessing the two quad-Flash memories of the stm32h750 disco kit  in parallel. In this configuration the `flash-id ` property is ignored.

Reply:
--> This could be a interesting configuration to be tested with  set `flash-id = <2>;` writing different value in flash 1 and flash 2 and comparing the read data.

Reply:
no flash-id property is flash 0 by default in single flash mode = compatible with the HW layout to access the flash device on bank 1.

Reply:
@FRASTM You're right that the flash id can be set to 1 (bank 1) or 2 (bank 2), resulting in setting `FlashID` (`FSEL`) to respectively 0 or 1 in the driver:
```
static int flash_stm32_qspi_init(const struct device *dev)
{
...
#if DT_NODE_HAS_PROP(DT_NODELABEL(quadspi), flash_id)
	uint8_t qspi_flash_id = DT_PROP(DT_NODELABEL(quadspi), flash_id);

	HAL_QSPI_SetFlashID(&dev_data->hqspi,
			    (qspi_flash_id - 1) << QUADSPI_CR_FSEL_Pos);
#endif
...
```
This setting is indeed only taken into account when `DualFlash` (`DFM`) is 0:

<img width="532" alt="Scherm­afbeelding 2024-01-05 om 16 14 47" src="https://github.com/zephyrproject-rtos/zephyr/assets/5477573/74a8001c-719d-46a7-861a-efae36bd44aa">

However, I see no evidence for bank 2 not being accessible in single flash mode.

Reply:
I've updated the PR again to add flash-id=1. In a separate PR we could have a look at propagating the DualFlash property to to the driver or even device tree to enable dual flash mode.